client/me/reauth-required/index.jsx

```diff
@@ -239,21 +223,70 @@ const ReauthRequired = createReactClass( {
 
 					{ this.renderSMSResendThrottled() }
 
-					<FormButtonsBar>
-						<FormButton
-							disabled={ this.state.validatingCode || ! this.preValidateAuthCode() }
-							onClick={ this.getClickHandler( 'Submit Validation Code on Reauth Required' ) }
-						>
-							{ this.props.translate( 'Verify' ) }
-						</FormButton>
-
-						{ this.renderSendSMSButton() }
-					</FormButtonsBar>
+					<FormButton
+						className="reauth-required__button"
+						disabled={ this.state.validatingCode || ! this.preValidateAuthCode() }
+						onClick={ this.getClickHandler( 'Submit Validation Code on Reauth Required' ) }
+					>
+						{ this.props.translate( 'Verify' ) }
+					</FormButton>
 				</form>
+			</Card>
+		);
+	},
+
+	renderSecurityKey() {
+		return (
+			<SecurityKeyForm
+				loginUserWithSecurityKey={ () => {
+					return this.props.twoStepAuthorization.loginUserWithSecurityKey( {
+						user_id: this.props.currentUserId,
+					} );
+				} }
+			/>
+		);
+	},
+
+	render: function() {
+		const method = this.props.twoStepAuthorization.isTwoStepSMSEnabled() ? 'sms' : 'authenticator';
+		const isSecurityKeySupported =
+			this.props.twoStepAuthorization.isSecurityKeyEnabled() && supported();
+
+		return (
+			<Dialog
+				autoFocus={ false }
+				className="reauth-required__dialog"
+				isFullScreen={ false }
+				isVisible={ this.props.twoStepAuthorization.isReauthRequired() }
+				buttons={ null }
+				onClose={ null }
+			>
+				{ isSecurityKeySupported &&
+					this.state.twoFactorAuthType === 'webauthn' &&
+					this.renderSecurityKey() }
+				{ this.state.twoFactorAuthType !== 'webauthn' && this.renderVerificationForm() }
+				<TwoFactorActions
+					twoFactorAuthType={ this.state.twoFactorAuthType }
+					onChange={ this.handleAuthSwitch }
+					isSmsSupported={ method === 'sms' || method === 'authenticator' }
```

Comment:
This is to avoid breaking the current behavior. That is, that even if your account is set up using an authenticator application, you can still request the codes to be sent by SMS. There's some more details in #36809